src/transformers/models/swiftformer/modeling_tf_swiftformer.py

```diff
@@ -0,0 +1,911 @@
+
+        self.scale_factor = dim**-0.5
+        self.proj = tf.keras.layers.Dense(dim, name="proj")
+        self.final = tf.keras.layers.Dense(dim, name="final")
+
+    def build(self, input_shape=None):
+        self.w_g = self.add_weight(
+            name="w_g",
+            shape=(self.dim, 1),
+            initializer=tf.keras.initializers.RandomNormal(mean=0, stddev=1),
+            trainable=True,
+        )
+
+        if self.built:
+            return
+        self.built = True
+        if getattr(self, "to_query", None) is not None:
+            with tf.name_scope(self.to_query.name):
+                self.to_query.build(self.dim)
+        if getattr(self, "to_key", None) is not None:
+            with tf.name_scope(self.to_key.name):
+                self.to_key.build(self.dim)
+        if getattr(self, "proj", None) is not None:
+            with tf.name_scope(self.proj.name):
+                self.proj.build(self.dim)
+        if getattr(self, "final", None) is not None:
+            with tf.name_scope(self.final.name):
+                self.final.build(self.dim)
+
+    def call(self, x: tf.Tensor) -> tf.Tensor:
+        query = self.to_query(x)
+        key = self.to_key(x)
+
+        query = tf.math.l2_normalize(query, dim=-1)
+        key = tf.math.l2_normalize(key, dim=-1)
+
+        query_weight = query @ self.w_g
+        scaled_query_weight = query_weight * self.scale_factor
+        scaled_query_weight = tf.nn.softmax(scaled_query_weight, axis=-1)
+
+        global_queries = tf.math.reduce_sum(scaled_query_weight * query, axis=1)
+        global_queries = tf.tile(tf.expand_dims(global_queries, 1), (1, key.shape[1], 1))
+
+        out = self.proj(global_queries * key) + query
+        out = self.final(out)
+
+        return out
+
+
+class TFSwiftFormerLocalRepresentation(tf.keras.layers.Layer):
+    """
+    Local Representation module for SwiftFormer that is implemented by 3*3 depth-wise and point-wise convolutions.
+
+    Input: tensor of shape `[batch_size, channels, height, width]`
+
+    Output: tensor of shape `[batch_size, channels, height, width]`
+    """
+
+    def __init__(self, config: SwiftFormerConfig, dim: int, **kwargs):
+        super().__init__(**kwargs)
+
+        self.dim = dim
+
+        self.pad = tf.keras.layers.ZeroPadding2D(padding=(1, 1))
+        self.depth_wise_conv = tf.keras.layers.Conv2D(dim, kernel_size=3, groups=dim, name="depth_wise_conv")
+        self.norm = tf.keras.layers.BatchNormalization(
+            epsilon=config.batch_norm_eps, momentum=0.9, name="norm"
+        )  # FIXME: momentum
+        self.point_wise_conv1 = tf.keras.layers.Conv2D(dim, kernel_size=1, name="point_wise_conv1")
+        self.act = get_tf_activation("gelu")
+        self.point_wise_conv2 = tf.keras.layers.Conv2D(dim, kernel_size=1, name="point_wise_conv2")
+        self.drop_path = tf.keras.layers.Identity(name="drop_path")  # FIXME: is this correct?
```

Comment:
Same here - I'd expect there to be a config value to control this

Reply:
https://github.com/huggingface/transformers/pull/23342/commits/2b3a76a0c9109bfc53316b6d09577d8f3e6632d8 addressed here

Reply:
My bad, that commit addressed this comment: https://github.com/huggingface/transformers/pull/23342#discussion_r1453858622

Reply:
Should I add a config parameter for each of these layers or should I have a general dropout rate for these layers?

Reply:
Hey @joaocmd, sorry for the delay in handling this! My suspicion is that `drop_path` will probably misbehave when we compile networks in TF (it will also likely cause problems when compiling with Torch 2.X). I think whatever you can do with it that works is fine - but just setting the drop_path layers to `Identity` is completely valid imo.